### PR TITLE
Issue creation command

### PR DESCRIPTION
**Pull request by luo-yuyuan**
#433 

Example usage: .issue apertium/phenny myGithubHandle myPassword This is the title of issue

This has been tested on a private channel.

EDIT: This does create the issue under your account.

### CODE REVIEW

File: modules/git.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 from threading import Thread
 import time
 import urllib.parse
+import requests
```
Comment (sushain97) [inline modules/git.py]: Please use the web module for web requests unless it isn't possible.

**Comment by luo-yuyuan** [comment]
I did not use ApertiumBot, however, since I didn't think it was necessary; but I don't know if we want to include it?

**Comment by sushain97** [comment]
Don't forget to write tests!

**Comment by luo-yuyuan** [comment]
@sushain97 what github user/repo should I use to test the valid case?

**Comment by sushain97** [comment]
It doesn't matter. The request shouldn't actually be sent out, we should just make sure the request is formed correctly.

File: modules/git.py (hunk 1 above)
Comment (luo-yuyuan) [reply modules/git.py]: Er, just curious - what is the reason that we prefer web over requests? Thanks!

Comment (sushain97) [reply modules/git.py]: web just wraps requests with some extra phenny specific stuff.

Comment (luo-yuyuan) [reply modules/git.py]: Ah, I see. I thought you were referring to the webpy framework! Thanks for the clarification :)

File: modules/git.py [hunk 2]
```diff
@@ -593,3 +593,34 @@ def retrieve_commit(phenny, input):
...
+
+def post_issue(phenny, input):
+    if len(input.bytes.split()) == 1:
+        phenny.say("Usage: .issue <org>/<repo> handle> <password> <issue title>")
+        return
+    elif len(input.bytes.split()) < 5:
+        phenny.say("Please try again. Usage: .issue <org>/<repo> handle> <password> <issue title>")
+        return
+
+    ret = make_issue(input.bytes.split()[1], input.bytes.split()[2], input.bytes.split()[3], " ".join(input.bytes.split()[4:]))
```
Comment (Androbin) [inline modules/git.py]: This line doesn't exactly look pragmatic to me.

File: modules/git.py [hunk 3]
```diff
@@ -593,3 +593,34 @@ def retrieve_commit(phenny, input):
...
+    if r.status_code == 201:
+        return 0
+    return 1
+
+def post_issue(phenny, input):
+    if len(input.bytes.split()) == 1:
+        phenny.say("Usage: .issue <org>/<repo> handle> <password> <issue title>")
+        return
+    elif len(input.bytes.split()) < 5:
+        phenny.say("Please try again. Usage: .issue <org>/<repo> handle> <password> <issue title>")
```
Comment (Androbin) [inline modules/git.py]: You don't want your password out in the wild on IRC.
You'd keep that in the configuration file and check `input.admin`.
You could also get a developer token from GitHub specifically with issues permissions.

**Comment by Androbin** [comment]
Of course you're free to test it with a dummy repo first.

File: modules/git.py (hunk 3 above)
Comment (sushain97) [reply modules/git.py]: oh no! begiak should only hold a token with issues permissions. no passwords anywhere

**Comment by wei2912** [comment]
Should we allow the sending of user passwords to begiak by PM? Seems like a security hazard.

**Comment by sushain97** [comment]
Nooo! No passwords anywhere.

File: modules/test/test_git.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    @catch_timeout
+    def test_issue_invalidPassword(self):
+        self.input.group = lambda x: [None, 'luo-yuyuan/phenny luo-yuyuan wrongpassword Issue Title'][x]
+        git.post_issue(self.phenny, self.input)
+        self.phenny.say.assert_called_once_with(
+            "Please try again. Usage: .issue <org>/<repo> <handle> <password> <issue title>")
+
+    @catch_timeout
+    def test_issue_validPassword(self):
+        self.input.group = lambda x: [None, 'luo-yuyuan/phenny luo-yuyuan 99bcc58e1642c398ac1ee420696c1356d0386786 Issue Title'][x]
```
Comment (sushain97) [inline modules/test/test_git.py]: Remove any sensitive tokens like this ASAP and delete the token from your GitHub account. I recommend force pushing over this to delete the commit.

Comment (sushain97) [reply modules/test/test_git.py]: (people scrape GitHub repos to find tokens like this one and then hack accounts using them)

**Comment by wei2912** [comment]
In that case, will all issues be created under ApertiumBot? If so, the creation of issues should be limited to a few people and the creator should be tagged by his/her handle.

**Comment by sushain97** [comment]
> In that case, will all issues be created under ApertiumBot? If so, the creation of issues should be limited to a few people and the creator should be tagged by his/her handle.

Yes, they will all be created by ApertiumBot who will say who really created them.

There's no need to limit this functionality. Anyone can go and create a ticket themselves. If it gets abused, then we can see about limiting it.

File: modules/git.py [hunk 5]
```diff
@@ -610,17 +610,25 @@ def make_issue(repo, username, password, title):
...
-        phenny.say("Please try again. Usage: .issue <org>/<repo> handle> <password> <issue title>")
+
+    command_split = command.split(' ')
+    if len(command_split) < 4:
+        phenny.say('Please try again. Usage: .issue <org>/<repo> handle> <password> <issue title>')
         return
 
-    ret = make_issue(input.bytes.split()[1], input.bytes.split()[2], input.bytes.split()[3], " ".join(input.bytes.split()[4:]))
+    command_spec = command_split[:3]
+    command_spec.append(" ".join(command_split[3:]))
```
Comment (Androbin) [inline modules/git.py]: Passing `maxsplit=4` to `command.split(' ')` would take care of these two lines for you. Just saying.

**Pull request closed without merging**